src/Persistence/PaYaml/Serialization/PaYamlSerializer.cs

```diff
@@ -153,14 +154,15 @@ public static bool CheckIsSequence(StringReader reader)
     {
         ArgumentNullException.ThrowIfNull(reader);
 
-        var builder = new DeserializerBuilder()
-            .IgnoreUnmatchedProperties()
-            .WithTypeConverter(new YamlSequenceTesterConverter());
-        var deserializer = builder.Build();
         try
         {
-            var results = deserializer.Deserialize<object[]>(reader);
-            return results is not null;
+            var parser = new Parser(reader);
+
+            // Need to consume the StreamStart and DocumentStart events to get to the root sequence
+            parser.TryConsume<StreamStart>(out _);
+            parser.TryConsume<DocumentStart>(out _);
+
+            return parser.TryConsume<SequenceStart>(out _);
```

Comment:
Hmm.   This would certainly be faster for checking if the start of string is a sequence (what I'd expect to see maybe 99.9% of the time), but wouldn't this report a false positive claiming that the following is a sequence?
```yaml
- foo
- bar
asdf: 'jkl;'
```

Reply:
yes. I was discussing this with @joem-msft . I guess the point of the method is to detect a potential sequence, not to validate the correctness of the yaml fragment. That would be left to the deserializer later on.

I'm open to suggestions though. The previous method is slower (even though we're talking microseconds here...), but safer.

Reply:
the above yaml is invalid, 'asdf' is a mapping key, but the previous lines indicate a sequence.
This scenario is out of scope of the utility. We only want to do the quickest check on the input to determine whether it looks like a sequence.

calling code will still need to deserialize the input to a type.

Reply:
Yeah that string is invalid yaml, but it is a possible fragment to receive by someone copy/pasting but missing the first line, like if the full string was something like 
```yaml
properties:
- foo
- bar
asdf: 'jkl;'
```
or 
```yaml
button1:
  properties:
  - foo
  - bar
  asdf: 'jkl;'
```

but if we're doing other validation for the invalid yaml later down the chain, then the point is moot.  ¯\\\_(ツ)_/¯